airflow/providers/amazon/aws/operators/batch.py

```diff
@@ -143,7 +151,7 @@ def __init__(
         region_name: str | None = None,
         tags: dict | None = None,
         wait_for_completion: bool = True,
-        deferrable: bool = False,
+        deferrable: bool = DEFAULT_DEFERRABLE,
```

Comment:
Not suggesting a change for this PR, but, wondering if we need to add a pre-commit hook to check that the default value for this param is set to DEFAULT_DEFERRABLE so that someone who is not aware of this change knows about it when they add new deferrable support to other operators/sensors.?

Reply:
Yes, I think that would be really helpful for contributors to know this feature and we might be able to migrate them to `BaseDeferrableOperator` all at once when min airflow version bumped to 2.7